src/lib.rs

```diff
@@ -235,6 +262,18 @@ impl Surface {
         })
     }
 
+    pub fn resize(&mut self, width: u32, height: u32) {
+        self.surface_impl.resize(width, height);
+    }
+
+    pub fn buffer_mut(&mut self) -> &mut [u32] {
```

Comment:
What should occur if someone tries to acquire the buffer before the size was set.

Reply:
Yeah, I guess that isn't handled correctly at present. Though it's also an issue if `resize` is called with a size of `0`.

We could define the crate's API to initialize size to `(1, 1)`, and make `resize` take `NonZeroU32` ([Glutin seems to use this type](https://docs.rs/glutin/latest/glutin/surface/trait.GlSurface.html#tymethod.resize)). Or not have a default and require size arguments to `Surface::new`, but that's might be inconvenient if it's common to pass `1` and resize later.

Or we could allow size `0`, but just return a buffer of `&mut []`, and make present a no-op (may be unexpected if you've already set a buffer though; it could present a 1x1 buffer that's just white/alpha to at least clear the window). 

So this is a decision to make about the public API, and we can adjust the backends to match.